src/mode/graphics.rs

```diff
@@ -194,19 +199,31 @@ where
 #[cfg(feature = "graphics")]
 extern crate embedded_graphics;
 #[cfg(feature = "graphics")]
-use self::embedded_graphics::{drawable, pixelcolor::PixelColorU8, Drawing};
+use self::embedded_graphics::{drawable, pixelcolor::BinaryColor, Drawing};
+#[cfg(feature = "graphics")]
+use core::convert::TryInto;
 
 #[cfg(feature = "graphics")]
-impl<DI> Drawing<PixelColorU8> for GraphicsMode<DI>
+impl<DI> Drawing<BinaryColor> for GraphicsMode<DI>
 where
     DI: DisplayInterface,
 {
     fn draw<T>(&mut self, item_pixels: T)
     where
-        T: Iterator<Item = drawable::Pixel<PixelColorU8>>,
+        T: IntoIterator<Item = drawable::Pixel<BinaryColor>>,
     {
-        for pixel in item_pixels {
-            self.set_pixel((pixel.0).0, (pixel.0).1, pixel.1.into_inner());
+        // Filter out pixels that are off the top left of the screen
+        let on_screen_pixels = item_pixels
+            .into_iter()
+            .filter(|drawable::Pixel(point, _)| point.x >= 0 && point.y >= 0);
+
+        for drawable::Pixel(point, color) in on_screen_pixels {
+            // NOTE: The filter above means the coordinate conversions should never panic
+            self.set_pixel(
+                point.x.try_into().expect("Point X coordinate is negative"),
```

Comment:
Using `try_into` here will probably check for `x >= 0` again, which is unnecessary and causes a slight runtime penalty.

Reply:
This was suggested by the compiler, but it can be simplified seeing as we do the positivity checks beforehand anyway.